cmd/registry/controller/controller.go

```diff
@@ -195,9 +195,21 @@ func generateUpdateActions(
 		return nil, nil, err
 	}
 
+	includeRevisions := strings.Contains(resourcePattern, "@")
+
 	// Iterate over a list of existing target resources to generate update actions
 	for _, targetResource := range resourceList {
-		visited[targetResource.ResourceName().ParentName().String()] = true
+		pn := targetResource.ResourceName().ParentName()
+
+		// if the pattern doesn't explicitly ask for revisions, ignore them
+		if !includeRevisions {
+			switch r := pn.(type) {
+			case patterns.SpecName:
+				r.RevisionID = ""
+				pn = r
+			}
+		}
+		visited[pn.String()] = true
```

Comment:
I think what we need here is ignoring the revision in the parent name while generating the filter (since the filter applies only on the resource name). 

So instead of this, we can make the change in `excludeVisitedParents` like this:
```
func excludeVisitedParents(v map[string]bool) string {
	// Wrap each string with quotes and join them with commas to build a JSON string array.
        // Ignore revision names if any exist in the parent names.
	jsonStrings := make([]string, 0, len(v))
	for parent := range v {
		jsonStrings = append(jsonStrings, fmt.Sprintf("%q", strings.Split(parent, "@")[0]))
	}
	return fmt.Sprintf("[%s].all(parent, !(name==parent))", strings.Join(jsonStrings, ","))
}
```

Thoughts?

Reply:
Yes, that will work - as long as we're ok dropping the guard to check for revision in the pattern, which we don't need now anyway.

Reply:
There is a function to validate the manifest definition: https://github.com/apigee/registry/blob/main/cmd/registry/controller/parser.go#L38

`ParseResource()` doesn't support `SpecRevisions` right now, so that guard is still in place, if we decide to add the specrevision support later, we can modify the validate function accordingly.